### PR TITLE
Editable Education Component

### PR DESCRIPTION
**Pull request by anirudhmungre**
closes: #88 
- Education Component Addable and Editable

### CODE REVIEW

File: front/src/components/App/KnownComponents/EducationComponent.vue [hunk 1]
```diff
@@ -50,11 +124,11 @@
                         </tr>
                         <tr>
                             <td>
-                                {{school.gradyear}}
+                                {{school.gradYear}}
                             </td>
                         </tr>
                     </table>
-                    <hr v-if="!(i == data.education.length -1)">
+                    <hr v-if="!(i === data.education.length -1)">
```
Comment (ethanelliott) [inline front/src/components/App/KnownComponents/EducationComponent.vue]: Logic could be simplified to `length - i > 1`

File: front/src/components/App/KnownComponents/EducationComponent.vue [hunk 2]
```diff
@@ -72,20 +146,42 @@
...
+        },
         methods: {
             cancel() {
                 this.dialog = false;
                 // console.log()
             },
             save() {
                 this.dialog = false;
-                // console.log()
+                this.data.education = Array.from(this.pEducation);
```
Comment (ethanelliott) [inline front/src/components/App/KnownComponents/EducationComponent.vue]: If I remember correctly, Array.from() won't properly deep-copy objects inside the array, so its probably better to stringify and parse for now, or use the `deepcopy` library that is already in the package.json

File: front/src/components/App/KnownComponents/EducationComponent.vue [hunk 3]
```diff
@@ -72,20 +146,42 @@
         },
         data: () => ({
             dialog: false,
+            pEducation: []
         }),
+        watch: {
+            dialog(val) {
+                if (val) {
```
Comment (ethanelliott) [inline front/src/components/App/KnownComponents/EducationComponent.vue]: Make sure this won't fail if the education data doesn't exist (like when the component is first added)

File: front/src/main.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ import 'material-design-icons-iconfont/dist/material-design-icons.css'
 
 LogRocket.init('ayfxcp/prtflio');
 
-const PROD = true;
+const PROD = false;
```
Comment (ethanelliott) [inline front/src/main.ts]: whoops prod should be true for the server

File: front/src/components/App/KnownComponents/EducationComponent.vue [hunk 5]
```diff
@@ -72,20 +146,42 @@
...
+                    this.pEducation = JSON.parse(JSON.stringify(this.data.education));
```
Comment (ethanelliott) [inline front/src/components/App/KnownComponents/EducationComponent.vue]: I think moving everything to `deepcopy` might be a good idea

File: front/src/components/App/KnownComponents/EducationComponent.vue (hunk 3 above)
Comment (anirudhmungre) [reply front/src/components/App/KnownComponents/EducationComponent.vue]: wont fail out, just defaults to blank strings

File: front/src/main.ts (hunk 4 above)
Comment (anirudhmungre) [reply front/src/main.ts]: whoops

**Pull request merged**